### ISSUE TITLE
Customizing Jackson configuration for HTTP Client

### ISSUE THREAD

**Issue by morki** [micronaut-projects/micronaut-core#3579]
We use many 3rd party HTTP services in our application. Some of those services are using snake case strategy, some camel case etc. Now we have to annotate every DTO class and enum with `@JsonNaming` annotation. It is very verbose and error prone.
Another options that are repeating many times in single client are `WRITE_DATES_AS_TIMESTAMPS` and date fotmats.

Maybe we could somehow add `JacksonConfiguration` to `HttpClientConfiguration`?
And maybe there is better solution already in Micronaut that I can't find.

**Comment by morki** [comment]
It seems that `JsonNaming` is even not working in combination with `@Introspected`, it is ignored.

**Comment by re-thc** [comment]
https://docs.micronaut.io/latest/guide/index.html#clientJackson does the @JacksonFeatures annotation not help for this case?

**Comment by morki** [comment]
@hc-codersatlas I think naming strategy is not (de)serialization feature, is it? I can't find any documentation about this, it is always configured separately, even in `application.yml`.

```yaml
jackson:
    propertyNamingStrategy: SNAKE_CASE
    serialization:
        indentOutput: true
        writeDatesAsTimestamps: false
    deserialization:
        useBigIntegerForInts: true
        failOnUnknownProperties: false
```

**Comment by morki** [comment]
It seems to be fixed by #5233
